src/token-builder/types.js

```diff
@@ -0,0 +1,192 @@
+const ALLOWED_ALGOS = new Set(['sha256', 'sha1', 'md5']);
+const ALLOWED_ENCODINGS = new Set(['hex', 'base64', 'base64url', 'base64percent']);
+
+export const REPLACE_CHAR_LIMIT = 100;
+
+export class TokenBase {
+  constructor({ name, cacheOverride = null, skipCache = false }) {
+    this.name = name;
+    this.type = 'base';
+    this.requiredProperties = ['name'];
+    this.errors = [];
+    this.cacheOverride = cacheOverride;
+    this.skipCache = skipCache;
+  }
+
+  validateOptions() {
+    const missingProps = [];
+    this.requiredProperties.forEach((prop) => {
+      if (!this[prop]) {
```

Comment:
Since `value` is the only property where an empty string is valid, I opted to keep this logic in place for `TokenBase`, but update how the validations are done for the `Replace` and `ReplaceLarge` tokens.